src/components/Reports/DonationsReport/DonationsReportTable.tsx

```diff
@@ -1,10 +1,242 @@
+
+  const title = time.monthLong + ' ' + time.year;
+
+  const hasNext = time.hasSame(DateTime.now().startOf('month'), 'month');
+
+  const setPrevMonth = () => {
+    setTime(time.minus({ months: 1 }));
+  };
+
+  const setNextMonth = () => {
+    setTime(time.plus({ months: 1 }));
+  };
+
+  const totalDonations = data.reduce((total, current) => {
+    return total + current.convertedAmount;
+  }, 0);
+
+  const calculateForeignTotals = () => {
+    const foreignTotals: {
+      [name: string]: { convertedTotal: number; foreignTotal: number };
+    } = {};
+    data.forEach((donation) => {
+      if (foreignTotals[donation.foreignCurrency] !== undefined) {
+        foreignTotals[donation.foreignCurrency].foreignTotal +=
+          donation.foreignAmount;
+        foreignTotals[donation.foreignCurrency].convertedTotal +=
+          donation.convertedAmount;
+      } else {
+        foreignTotals[donation.foreignCurrency] = {
+          convertedTotal: donation.convertedAmount,
+          foreignTotal: donation.foreignAmount,
+        };
+      }
+    });
+
+    return foreignTotals;
+  };
+
+  const foreignTotals = calculateForeignTotals();
+
+  return (
+    <>
+      <Box style={{ display: 'flex', margin: 8 }}>
+        <Typography variant="h6">{title}</Typography>
+        <Button
+          style={{ marginLeft: 'auto' }}
+          variant="contained"
+          startIcon={<ChevronLeftIcon />}
+          size="small"
+          onClick={() => setPrevMonth()}
+        >
+          Previous Month
+        </Button>
+        <Button
+          variant="contained"
+          endIcon={<ChevronRightIcon />}
+          size="small"
+          onClick={() => setNextMonth()}
+          disabled={hasNext}
+        >
+          Next Month
+        </Button>
+      </Box>
+      <Divider style={{ margin: 12 }} variant="middle"></Divider>
+      {!isEmpty ? (
+        <Data>
+          <DataGrid
+            rows={data}
+            columns={columns}
+            autoPageSize
+            autoHeight
+            disableSelectionOnClick
+            hideFooter
+          />
+          <Table>
+            {Object.entries(foreignTotals).map(([currency, total]) => (
+              <TableRow key={currency}>
+                <TableCell style={{ width: 375 }}>
+                  <Typography style={{ float: 'right', fontWeight: 'bold' }}>
+                    {t('Total ' + currency + ' Donations:')}
+                  </Typography>
+                </TableCell>
+                <TableCell style={{ width: 160 }}>
+                  <Typography style={{ float: 'left', fontWeight: 'bold' }}>
+                    {' ' + total.convertedTotal + ' ' + accountCurrency}
+                  </Typography>
+                </TableCell>
+                <TableCell style={{}}>
+                  <Typography style={{ float: 'left', fontWeight: 'bold' }}>
+                    {' ' + total.foreignTotal + ' ' + currency}
```

Comment:
Can you just use JSX for these instead of `+`? The left space should probably be done with CSS padding instead of extra characters.